src/controller/CHIPDeviceController.cpp

```diff
@@ -2092,7 +2080,37 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
     case CommissioningStage::kSecurePairing:
         break;
     }
-} // namespace Controller
+}
+
+CHIP_ERROR DeviceController::UpdateDevice(NodeId deviceId)
+{
+    VerifyOrReturnError(mState == State::Initialized && mFabricInfo != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    OperationalDeviceProxy * proxy = GetDeviceSession(mFabricInfo->GetPeerIdForNode(deviceId));
+    VerifyOrReturnError(proxy != nullptr, CHIP_ERROR_NOT_FOUND);
+
+    return proxy->LookupPeerAddress();
+}
+
+OperationalDeviceProxy * DeviceController::GetDeviceSession(const PeerId & peerId)
+{
+    return mCASESessionManager->FindExistingSession(peerId);
+}
+
+OperationalDeviceProxy * DeviceCommissioner::GetDeviceSession(const PeerId & peerId)
```

Comment:
The behaviour difference between the commissioner and the controller here is a bit confusing. I know there's a lot of plumbing here, but we might want to rename these functions to make the behaviour more clear. Perhaps we should consolidate these into the controller as "GetExistingOperationalSession", "FindOrEstablishOperationalSession" and add an option to force a DNS lookup on either one. Not for this PR perhaps, but this is one area where the API could use some cleaning before FC.

Reply:
I would like to do this as a followup. This is poorly named (but did not know what to name it) and also want to rename 'UpdateDevice' since that one is also unclear. Gathered a solid list of 'can do more' as part of this PR.